dev/npm.ts

```diff
@@ -24,6 +24,64 @@ const searchGenerator: Fig.Generator = {
   // },
 };
 
+const dependenciesGenerator: Fig.Generator = {
+  script: "cat package.json",
+  postProcess: function (out, context) {
+    if (out.trim() === "") {
+      return [];
+    }
+    try {
+      const packageContent = JSON.parse(out);
+      const dependencies = packageContent["dependencies"];
+      if (dependencies) {
+        const dps = Object.keys(dependencies);
+        return dps
+          .filter(function (pkg) {
+            let isListed = false;
+            for (let i = 3; i <= context.length; i++) {
+              if (pkg == context[i - 2]) {
+                isListed = true;
+              }
+            }
+            return !isListed;
```

Comment:
I think we can simplify this a bit, not tested but something like:
```suggestion
            const isListed = context.some((current) => current === pkg);
            return !isListed;
```

Reply:
It is also strange that the iteration was starting with `i = 3` 🤔

Reply:
> It is also strange that the iteration was starting with `i = 3` 🤔

I think it's because we want to skip the first two tokens, which are `npm <uninstall/remove/rm/r>`

Reply:
I thought at that but I believe it is fairly confusing.

Reply:
Wow @QuiiBz, great suggestion! I swear I learn something new every time you review my PRs, thanks for making me a better JS programmer! @fedeci yeah, I wanted to skip the first 2 tokens but agreed, it is definitely confusing. Replacing with Tom's suggestion :)

Reply:
Thanks for those kind words @Nutlope!